pandas/parser.pyx

```diff
@@ -548,6 +555,29 @@ cdef class TextReader:
     def set_error_bad_lines(self, int status):
         self.parser.error_bad_lines = status
 
+    def _set_quoting(self, quote_char, quoting):
+        if not isinstance(quoting, int):
+            raise TypeError('"quoting" must be an integer')
+
+        if not 0 <= quoting <= 3:
```

Comment:
Actually, I take that back.  The current checks will align nicely with what the Python engine currently does in its `csv` library.  I do check that it is a valid value by first checking whether or not it's an integer.  Then if so, it can only take on four values (between 0 and 3 inclusive, hence my subsequent check.